airflow/models/taskinstance.py

```diff
@@ -418,7 +421,7 @@ def error(self, session=None):
         session.commit()
 
     @provide_session
-    def refresh_from_db(self, session=None, lock_for_update=False, refresh_executor_config=False):
+    def refresh_from_db(self, session=None, lock_for_update=False, refresh_executor_config=False) -> None:
```

Comment:
There is a subtle difference explained here: https://realpython.com/python-type-checking/#functions-without-return-values  

If you specify -> None, then you will get warning if you try to assign return value to a variable. If you don't - you get no warning (None is still a valid return value that you can assign to something). So I started to add it (but not consistently, True). 
I might review the changes and add -> None in all the affected non-return functions and try to do it in the future as well.

WDYT @feluelle ?

Reply:
If we do it consistently then I am fine with it (actually I like it more :D). Because it programming languages like Java or C# I got used to specify `void` for no return value.

So if we can somehow force this to be defined I would go for it. 👍

Reply:
Some time in the future :)